internal/io/seekable.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the LGPLv3, see COPYING and COPYING.LESSER file for details.
+
+package io
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"gopkg.in/goose.v2/errors"
+)
+
+// maxBufSize holds the maximum amount of data
+// that may be allocated as a buffer when sending
+// a request when the body is not seekable.
+const maxBufSize = 1024 * 1024 * 1024
```

Comment:
This is still quite large... I wonder if we should make buffering scheme pluggable. Something for later.